asm_utils/src/data_parser.rs

```diff
@@ -14,76 +18,131 @@ pub enum DataValue {
 
 impl DataValue {
     /// Returns the size of the value in bytes.
-    pub fn size(&self) -> usize {
+    ///
+    /// The address is necessary because the size of the alignment padding
+    /// depending on what address it is defined on.
+    pub fn size(&self, from_addr: usize) -> usize {
         match self {
             DataValue::Direct(data) => data.len(),
             DataValue::Zero(length) => *length,
+            DataValue::Alignment(bytes) => alignment_size(from_addr, *bytes),
             DataValue::Reference(_) => 4,
             DataValue::Offset(..) => 4,
         }
     }
 }
 
+struct DataObjects<'a> {
+    objects: BTreeMap<String, Vec<DataValue>>,
+    object_order: Vec<String>,
+    current_label: Option<&'a str>,
+}
+
+impl<'a> DataObjects<'a> {
+    fn new() -> Self {
```

Comment:
Default::default?

Reply:
This should even be `#[derive(Default)]`

Reply:
Shouldn't we implement traits as needed? Anyway, this current version, using this struct, is broken.